app/main.py

```diff
@@ -1,6 +1,14 @@
-from typing import Callable
+from collections.abc import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args) -> Callable:
+        if args not in storage:
+            print("Calculating new result")
+            return storage.setdefault(args, func(*args))
+        else:
+            print("Getting from cache")
+            return storage.get(args)
+    return wrapper
```

Comment:
Always add a blank line at the end of the file.

Reply:
Yeh, sure, but it is there (flake8 always push me to do this)